src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+ * Supported features are:
+ * <p>
+ * Monitoring:
+ * <p>
+ * Controlling:
+ *
+ * @author Harry
+ * @since 1.0.0
+ */
+public class HaivisionX4DecoderCommunicator extends RestCommunicator implements Monitorable, Controller {
+
+	private final String uuidDay = UUID.randomUUID().toString().replace(DecoderConstant.DASH, "");
+	private final String uuidSeconds = UUID.randomUUID().toString().replace(DecoderConstant.DASH, "");
+	private AuthenticationCookie authenticationCookie;
+	private AuthenticationInfo authenticationInfo;
+	private HashMap<String, String> failedMonitor;
+	private ObjectMapper objectMapper;
+
+	//Adapter Properties
+	private String streamName;
+	private String portNumber;
+	private String streamStatus;
+
+	/**
+	 * Retrieves {@code {@link #streamName }}
+	 *
+	 * @return value of {@link #streamName}
+	 */
+	public String getStreamName() {
+		return streamName;
+	}
+
+	/**
+	 * Sets {@code streamsName}
+	 *
+	 * @param streamsName the {@code java.lang.String} field
+	 */
+	public void setStreamName(String streamsName) {
+		this.streamName = streamsName;
+	}
+
+	/**
+	 * Retrieves {@code {@link #portNumber}}
+	 *
+	 * @return value of {@link #portNumber}
+	 */
+	public String getPortNumber() {
+		return portNumber;
+	}
+
+	/**
+	 * Sets {@code portNumber}
+	 *
+	 * @param portNumber the {@code java.lang.String} field
+	 */
+	public void setPortNumber(String portNumber) {
+		this.portNumber = portNumber;
+	}
+
+	/**
+	 * Retrieves {@code {@link #streamStatus}}
+	 *
+	 * @return value of {@link #streamStatus}
+	 */
+	public String getStreamStatus() {
+		return streamStatus;
+	}
+
+	/**
+	 * Sets {@code streamStatus}
+	 *
+	 * @param streamStatus the {@code java.lang.String} field
+	 */
+	public void setStreamStatus(String streamStatus) {
+		this.streamStatus = streamStatus;
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param controllableProperty ControllableProperty instance
+	 * @throws Exception if unexpected the value
+	 */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+		// To do
+	}
+
+	/**
+	 * This method is called by Symphony to control the properties in the device
+	 *
+	 * @param list the list ControllableProperty instance
+	 * @throws Exception if ControllableProperty list is empty
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) throws Exception {
+		// To do
+	}
+
+	/**
+	 * This method is called by Symphony to get the list of statistics to be displayed
+	 *
+	 * @return List<Statistics> This return the list of statistics
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		Map<String, String> stats = new HashMap<>();
+		failedMonitor = new HashMap<>();
+		authenticationCookie = initAuthenticationCookie();
```

Comment:
please directly write authenticationCookie = new AuthenticationCookie(); instead of the new method initAuthenticationCookie. btw, do we need to re-init AuthenticationCookie every interval ? If we can reuse it, please check if authenticationCookie == null -> new AuthenticationCookie, otherwise, do not thing